### PR TITLE
uf-8632: Removed the DB and added a caffeine cache.

### PR DESCRIPTION
**Pull request by mha109**

### CODE REVIEW

File: src/main/java/se/sundsvall/businessengagements/config/CacheConfig.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+public class CacheConfig {
+
+	private final CacheProperties cacheProperties;
+
+	public CacheConfig(CacheProperties cacheProperties) {
+		this.cacheProperties = cacheProperties;
+	}
+
+	@Bean
+	public Caffeine<Object, Object> caffeineConfig() {
```
Comment (foobar-1337) [inline src/main/java/se/sundsvall/businessengagements/config/CacheConfig.java:27]: remove "public"?

File: src/main/java/se/sundsvall/businessengagements/config/CacheConfig.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+			.recordStats();
+	}
+
+	/**
+	 * Will configure a CacheManager if caching is enabled, otherwise a NoOpCacheManager (disabled cache) will be returned.
+	 * @param caffeine Caffeine configuration
+	 * @return CacheManager
+	 */
+	@Bean
+	public CacheManager cacheManager(Caffeine<Object, Object> caffeine) {
```
Comment (foobar-1337) [inline src/main/java/se/sundsvall/businessengagements/config/CacheConfig.java:40]: remove "public"?

File: src/main/java/se/sundsvall/businessengagements/config/CacheConfig.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+		if(cacheProperties.enabled()) {
+			var caffeineCacheManager = new CaffeineCacheManager(LEGAL_ID_CACHE, PARTY_ID_CACHE, BUSINESS_INFORMATION_CACHE, BUSINESS_ENGAGEMENTS_CACHE);
```
Comment (foobar-1337) [inline src/main/java/se/sundsvall/businessengagements/config/CacheConfig.java:42]: Isn't all this usually managed by properties instead?

File: src/main/resources/application-it.yaml [hunk 4]
```diff
@@ -30,3 +16,9 @@ integration:
     oauth2-client-secret: fake
     oauth2-token-url: http://localhost:${wiremock.server.port}/token
     url: http://localhost:${wiremock.server.port}/party
+# Cache properties
+appconfig:
+  cache:
+    enabled: false
```
Comment (foobar-1337) [inline src/main/resources/application-it.yaml:22]:
```
spring:
  cache:
    type: none
```

...will do the same, right?

File: src/main/java/se/sundsvall/businessengagements/config/CacheConfig.java [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+import org.springframework.cache.CacheManager;
+import org.springframework.cache.annotation.EnableCaching;
+import org.springframework.cache.caffeine.CaffeineCacheManager;
+import org.springframework.cache.support.NoOpCacheManager;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+@EnableCaching
+@Configuration
+public class CacheConfig {
```
Comment (foobar-1337) [inline src/main/java/se/sundsvall/businessengagements/config/CacheConfig.java:18]: Do you really need this class at all?

File: src/main/java/se/sundsvall/businessengagements/config/CacheProperties.java [hunk 6]
```diff
@@ -0,0 +1,13 @@
+package se.sundsvall.businessengagements.config;
+
+import java.time.Duration;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+
+@ConfigurationProperties(prefix = "appconfig.cache")
+public record CacheProperties(
```
Comment (foobar-1337) [inline src/main/java/se/sundsvall/businessengagements/config/CacheProperties.java:8]: Is this necessary?

File: src/main/java/se/sundsvall/businessengagements/config/CacheConfig.java (hunk 5 above)
Comment (mha109) [reply src/main/java/se/sundsvall/businessengagements/config/CacheConfig.java:18]: Somehow I messed up the config and it didn't work at first so I went this route.
Tried with only config and now it works. I'll remove these classes.

**Review by foobar-1337 — APPROVED** [review]

**Pull request merged**